express-zod-api/src/io-schema.ts

```diff
@@ -32,31 +33,66 @@ export const getFinalEndpointInputSchema = <
   ) as z.ZodIntersection<MIN, IN>;
 };
 
-export const extractObjectSchema = (subject: IOSchema): z.ZodObject => {
-  if (subject instanceof z.ZodObject) return subject;
-  if (subject instanceof z.ZodInterface) {
-    const { optional } = subject._zod.def;
-    const mask = R.zipObj(optional, Array(optional.length).fill(true));
-    const partial = subject.pick(mask);
-    const required = subject.omit(mask);
-    return z
-      .object(required._zod.def.shape)
-      .extend(z.object(partial._zod.def.shape).partial());
-  }
-  if (
-    subject instanceof z.ZodUnion ||
-    subject instanceof z.ZodDiscriminatedUnion
-  ) {
-    return subject._zod.def.options
-      .map((option) => extractObjectSchema(option as IOSchema))
-      .reduce((acc, option) => acc.extend(option.partial()), z.object({}));
-  }
-  if (subject instanceof z.ZodPipe)
-    return extractObjectSchema(subject.in as IOSchema);
-  if (subject instanceof z.ZodIntersection) {
-    return extractObjectSchema(subject._zod.def.left as IOSchema).extend(
-      extractObjectSchema(subject._zod.def.right as IOSchema),
-    );
+const isJsonObjectSchema = (
+  subject: JSONSchema.BaseSchema,
+): subject is JSONSchema.ObjectSchema => subject.type === "object";
+
+export const flattenIO = (jsonSchema: JSONSchema.BaseSchema) => {
```

Comment:
it does the thing which is very similar to `intersect()` from documentation helpers.
should explore the possibility to DNRY

Reply:
made todo for another PR